### ISSUE TITLE
Player profile API broken on 1.15

### ISSUE THREAD

**Issue by MrIvanPlays** [PaperMC/Paper#2793]
### What behaviour is expected:
My skin to change

### What behaviour is observed:
My skin is default steve 

### Steps/models to reproduce:
Run this code
```java
    Player player = // you know what to do here
    PlayerProfile profile = player.getPlayerProfile();
    profile.setProperty(new ProfileProperty("textures", /* replace with skin texture */ , /* skin texture signature */));
    player.setPlayerProfile(profile);
```

You see the skin doesn't change. 

### Plugin list:
AnnouncementsEverywhere, Skins

### Paper build number:
[09:12:33 INFO]: This server is running Paper version git-Paper-29 (MC: 1.15.1) (Implementing API version 1.15.1-R0.1-SNAPSHOT)
[09:12:33 INFO]: You are running the latest version

### Anything else:
The Skins plugin is being a dev build (nowhere available except my pc cuz I don't wanna commit before everything is finished so I can keep the git history clean). Before you say it's something with my skin fetcher, you're wrong, as the build was before tested on spigot and worked properly, and before tested on paper nothing about it was touched.

**Comment by MiniDigger** [comment]
How can you test this on spigot when this is a paper API?

When exactly are you running this code? Like which event. If you run it after being spawend in, you need to respawn the player.

**Comment by MrIvanPlays** [comment]
> How can you test this on spigot when this is a paper API?

The java terminology of that is "abstraction". I have another end which is being used for spigot and craftbukkit. It directly touches the NMS, doesn't support dynamic skin change but on reconnect it changes the skin. 

> When exactly are you running this code? Like which event. If you run it after being spawend in, you need to respawn the player.

It is PlayerJoinEvent, but that really doesn't matter in that case as the code is also being ran in a command and again: applies nothing to the player skin. 

EDIT: If you don't know the skins plugin is open source and its source code is available at my github [here](https://github.com/MrIvanPlays/Skins). (before you start blaming me for license just stfu I'm too lazy to change it to MIT again)

Any improvements to [this](https://github.com/MrIvanPlays/Skins/tree/master/plugin/bukkit/abstraction) are being accepted (everything other I've improved in my dev build).

**Comment by electronicboy** [comment]
![image](https://user-images.githubusercontent.com/1228900/71412082-fac37800-2643-11ea-96b2-d7bc3d1d42d1.png)

`The java terminology of that is "abstraction".` aka, completely irrelevant, as the code you're flashing off here is literally irrelevant to your implementation elsewhere

**Comment by MrIvanPlays** [comment]
Ok. So it looks like this is a problem with something other and not the player profile api. It looks like when I was testing the dev build on spigot, i was using outdated version of spigot, now I used latest version, and it sets my skin on steve. At current version the plugin is using `.join()` on the futures, which makes everything sync, now in the dev build where it doesn't use `.join()` it uses `.whenComplete()` and for some reason this breaks the skin set. Perhaps in that `.whenComplete()` I should throw the bukkit scheduler in order for the skin to actually set on the player. 

Closing.
